static/script-tests/tests/devices/mediaplayer/samsung_streaming.js

```diff
@@ -0,0 +1,1978 @@
+                              function(application, MediaPlayerImpl, MediaPlayer) {
+                                  self._device = application.getDevice();
+                                  self.sandbox.stub(self._device, 'getScreenSize').returns(screenSize);
+                                  self._errorLog = self.sandbox.stub(self._device.getLogger(), 'error');
+                                  self._mediaPlayer = self._device.getMediaPlayer();
+            
+                                  action.call(self, MediaPlayer);
+                              }, config);
+    };
+
+    //---------------------
+    // Samsung Maple specific tests
+    //---------------------
+    
+    var listenerEventCodes = {
+        CONNECTION_FAILED : 1,
+        AUTHENTICATION_FAILED : 2,
+        STREAM_NOT_FOUND : 3,
+        NETWORK_DISCONNECTED : 4,
+        RENDER_ERROR : 6,
+        RENDERING_START : 7,
+        RENDERING_COMPLETE : 8,
+        STREAM_INFO_READY : 9,
+        DECODING_COMPLETE : 10,
+        BUFFERING_START : 11,
+        BUFFERING_COMPLETE : 12,
+        BUFFERING_PROGRESS : 13,
+        CURRENT_PLAYBACK_TIME : 14
+    };
+
+    this.SamsungStreamingMediaPlayerTests.prototype.testSamsungMapleListenerFunctionsAddedDuringSetSource = function(queue) {
+        expectAsserts(2);
+        runMediaPlayerTest(this, queue, function(MediaPlayer) {
+            
+            assertUndefined('Expecting playerPlaugin.OnEvent to be undefined', playerPlugin.OnEvent);
+
+            this._mediaPlayer.setSource(MediaPlayer.TYPE.VIDEO, 'testURL', 'video/mp4');
+
+            assertFunction('Expecting playerPlaugin.OnEvent to be function', playerPlugin.OnEvent);
+        });
+    };
+
+    this.SamsungStreamingMediaPlayerTests.prototype.testSamsungMapleListenerFunctionsRemovedOnTransitionToErrorState = function(queue) {
+        expectAsserts(2);
+        runMediaPlayerTest(this, queue, function(MediaPlayer) {
+
+            this._mediaPlayer.setSource(MediaPlayer.TYPE.VIDEO, 'testURL', 'video/mp4');
+
+            assertFunction('Expecting playerPlaugin.OnEvent to be function', playerPlugin.OnEvent);
+
+            try {
+                this._mediaPlayer.pause();
+            } catch (e) {}
+
+            assertUndefined('Expecting playerPlaugin.OnEvent to be undefined', playerPlugin.OnEvent);
+        });
+    };
+
+    this.SamsungStreamingMediaPlayerTests.prototype.testSamsungMapleListenerFunctionsRemovedOnReset = function(queue) {
+        expectAsserts(2);
+        runMediaPlayerTest(this, queue, function(MediaPlayer) {
+
+            this._mediaPlayer.setSource(MediaPlayer.TYPE.VIDEO, 'testURL', 'video/mp4');
+
+            assertFunction('Expecting playerPlaugin.OnEvent to be function', playerPlugin.OnEvent);
+
+            this._mediaPlayer.reset();
+
+            assertUndefined('Expecting playerPlaugin.OnEvent to be undefined', playerPlugin.OnEvent);
+        });
+    };
+
```

Comment:
The next 3 tests appear to test the same things as the previous 3, so I think 1 set should be removed.